### ISSUE TITLE
Missing antigen & serum information when reading ace files

### ISSUE THREAD

**Issue by drserajames** [acorg/Racmacs#81]
When I use Racmacs to read in ace files that I've downloaded from Eu's interactive maps, the antigen & serum date, id & sequence are "". But this information is in the ace file when opened with a plain text reader. Eu's specification for ace files is [here](https://github.com/acorg/acmacs-chart-2/blob/master/doc/ace-format.json)

**Comment by shwilks** [comment]
I'm slowly adding these, do we want all of them, I guess so?  If there are other particular priorities let me know, currently we the following are included

__Map attributes__
- [ ] "v": "virus, e.g. INFLUENZA (default, if omitted), HPV, generic, DENGE",
- [ ] "V": "virus type and subtype, e.g. B or A(H3N2) or serotype"
- [ ] "A": "assay: HI, FOCUST REDUCTION, PRNT",
- [ ] "D": "table/assay date, e.g. 20160602.002",
- [x ] "N": "user supplied name",
- [ ] "l": "lab",
- [ ] "r": "rbc species, e.g. turkey",
- [ ] "s": "subset/lineage, e.g. 2009PDM",
- [ ] "T": "table type: A(NTIGENIC) - default, G(ENETIC)"
- [ ] "S": []                    // source table info list, each entry is like "i"

__Antigen attributes__
- [x] "N": "mandatory name: TYPE(SUBTYPE)/[HOST/]LOCATION/ISOLATION/YEAR or CDC_ABBR NAME or UNRECOGNIZED NAME",
- [x] "D": "date, e.g. 2016-04-04",
- [ ] "L": "lineage: Y[AMAGATA] or V[ICTORIA]",
- [x] "P": "passage, e.g. MDCK2/SIAT1 (2016-05-12)",
- [ ] "R": "reassortant, e.g. NYMC-51C",
- [x] "l": ["CDC#2013706008"], // lab ids, e.g. CDC#2013706008",
- [ ] "S": "R", // string of single letter semantic boolean attributes: R - reference, E - egg, V - current vaccine, v - previous vaccine, S - vaccine surrogate
- [x] "a": ["DISTINCT", "HA-Y156T", "extra"], // annotations: DISTINCT for control duplicate, mutation information, unrecognized extra data preventing merging
- [x] "c": ["5.2.1"],                         // clades, either provided in the original table (e.g. H5) or extracted from sequence database
- [ ] "C": "NORTH-AMERICA",                   // continent: ASIA, AUSTRALIA-OCEANIA, NORTH-AMERICA, EUROPE, RUSSIA, AFRICA, MIDDLE-EAST, SOUTH-AMERICA, CENTRAL-AMERICA
- [x] "A": "",                                // aligned amino-acid sequence
- [ ] "B": "",                                // aligned nucleotide sequence

__Serum attributes__
- [x] "N": "mandatory name: [TYPE(SUBTYPE)/]{HOST/]LOCATION/ISOLATION/YEAR or UNRECOGNIZED NAME",
- [ ] "L": "lineage: Y[AMAGATA] or V[ICTORIA]",
- [x] "P": "passage, e.g. MDCK2/SIAT1 (2016-05-12)",
- [ ] "R": "reassortant, e.g. NYMC-51C",
- [x] "I": "serum id, e.g. CDC 2016-045"
- [ ] "S": "", // string of single letter semantic boolean attributes: E - egg
- [x] "c": ["5.2.1"],                         // clades by homologous antigen
- [ ] "h": [0],                 // homologous antigen indices
- [x] "a": ["BOOSTED", "CONC 2:1", "HA-Y156T", "extra"], // annotations: boosted, conc, mutation information, unrecognized extra data preventing merging
- [ ] "s": "serum species, e.g FERRET",
- [x] "A": "",                                // aligned amino-acid sequence
- [ ] "B": "",                                // aligned nucleotide sequence

**Comment by drserajames** [comment]
Are the ticks the ones you've done?

**Comment by shwilks** [comment]
Yes, although some only in v1.1.28 which is still just on my fork

**Comment by drserajames** [comment]
ID and passage would have been my faves, but you've already done them. In terms of the others, these are the ones I think will be most useful in the short term

antigen
- "R": recombinant antigens are plotted as a tilted egg
- "C": used in to plot the continent of origin in by month antigenic maps
- "S": used to show some antigens as larger circles on the map

serum
- "h": homologous antigen is useful calculating alternative distance metrics, and I'd need to to plot serum circles. 

My gut feeling that the others can be assumed or inferred. But I expect they all exist for a reason.

**Comment by drserajames** [comment]
Just remembered that serum species is sometime very useful (one WHOCC occasionally uses sheep sera!).

Perhaps you could leave some or all of these, show me how to fix one and the rest would be good practice for me?

**Comment by drserajames** [comment]
I see those two options are different, but don't have a good understanding on the implications. I think `mapRBCspecies()` etc would be easier to use on a list of maps to get the map information? But that might just be my limited R knowledge.

I must confess I'm not feeling super keen this week. If you show me how it works next week, I could give it a go.
